nvflare/app_common/utils/fl_model_utils.py

```diff
@@ -0,0 +1,120 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from nvflare.apis.dxo import DXO, DataKind, from_shareable
+from nvflare.apis.shareable import Shareable
+from nvflare.app_common.abstract.fl_model import FLModel, FLModelConst, TransferType
+from nvflare.app_common.app_constant import AppConstants
+
+MODEL_ATTRS = [
+    FLModelConst.OPTIMIZER,
+    FLModelConst.METRICS,
+    FLModelConst.CONFIGS,
+    FLModelConst.CLIENT_WEIGHTS,
+    FLModelConst.ROUND,
+    FLModelConst.META,
+]
+
+
+class FLModelUtils:
+    @staticmethod
+    def to_nvf(fl_model: FLModel) -> Shareable:
+        """From external FLModel to NVFlare side shareable."""
+        if fl_model.transfer_type == TransferType.MODEL:
+            data_kind = DataKind.WEIGHTS
+        elif fl_model.transfer_type == TransferType.MODEL_DIFF:
+            data_kind = DataKind.WEIGHT_DIFF
+        else:
+            raise ValueError("Invalid fl_model.")
+
+        dxo = DXO(data_kind, data=fl_model.model, meta={})
+        shareable = dxo.to_shareable()
+        if fl_model.round:
+            shareable.set_header(AppConstants.CURRENT_ROUND, fl_model.round)
+        if fl_model.meta:
+            if AppConstants.NUM_ROUNDS in fl_model.meta:
```

Comment:
model owner is no longer needed. It's only for the server side to use. It is in the cookie.
We could put num_of_rounds in both the shareable header and meta. It must be in shareable header; otherwise server side will break.
This is a dilemma: it is more precise to have everything defined in the structure, but we don't know "everything"; to be more flexible, we create a "catch-all" structure called "meta" or "config" or "extra", but then it's clear what to put there and what are the keys.
I think we should just define these keys (and maintain them in a std place) whenever we have a need.

Reply:
Yes. 
Another candidate for "meta" is "validate_type". I know we talked about removing it, but not sure that's possible.
For now, let's add a key and populate it into the meta to be compatible with what we have. We can remove it later if needed.

Reply:
https://github.com/NVIDIA/NVFlare/blob/dev/examples/advanced/cifar10/pt/learners/cifar10_learner.py#L413

^ put this is meta.